src/types/contextAction/index.ts

```diff
@@ -1,12 +1,17 @@
 import { HeuristicContextAction } from './heuristicContextAction';
-import { ProtocolContextAction } from './protocolContextAction';
+import { ProtocolContextAction, Protocols } from './protocolContextAction';
 import { BridgeContextAction } from './bridge';
 
 export type ContextAction =
   | HeuristicContextAction
   | ProtocolContextAction
   | BridgeContextAction;
 
+export type ContextActionFull =
```

Comment:
Maybe `ContextActionID` would be a more consistent name with the way we use id within variables of type `contextAction`
```suggestion
export type ContextActionID =
```